lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt

```diff
@@ -0,0 +1,28 @@
+package com.swmansion.starknet.data.types
+
+sealed class StarknetResult<out T> {
+
+    data class Success<T>(val value: T) : StarknetResult<T>()
+
+    data class Failure<T>(val exception: Exception) : StarknetResult<T>()
```

Comment:
Why not use Kotlin `Result` under the hood, and avoid implementing any logic ourselves?